### PR TITLE
Fixed tabs component

### PR DESCRIPTION
**Pull request by jeluard**
### Expected

Tabs from wallet transactions screen have same behavior than main tabs.

### Steps to test:
- Open Status
- Navigate to `wallet` / `transactions` screen
- Swipe between tabs and click on tabs
- COntent and titles are updated accordingly

status: ready

### CODE REVIEW

**Comment by flexsurfer** [comment]
@jeluard could you write Expected and Actual behavior

**Comment by jeluard** [comment]
@flexsurfer It was not working at all. Expected behavior is that you can both swipe and click on tabs while content and titles are properly updated.

File: src/status_im/components/tabs/views.cljs [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
+            [status-im.components.tabs.styles :as styles]
+            [status-im.utils.platform :as p]))
 
-(defn- tab [{:keys [view-id title icon-active icon-inactive selected-view-id]}]
+(defn- tab [{:keys [view-id title icon-active icon-inactive selected-view-id on-press]}]
   (let [active? (= view-id selected-view-id)]
     [react/touchable-highlight {:style    styles/tab
                                 :disabled active?
-                                :on-press  #(dispatch [:navigate-to-tab view-id])}
+                                :on-press  #(if on-press (do (println "tab click" view-id) (on-press view-id)) (dispatch [:navigate-to-tab view-id]))}
```
Comment (yenda) [inline src/status_im/components/tabs/views.cljs]: is it intentional or debug leftover ?

Comment (jeluard) [reply src/status_im/components/tabs/views.cljs]: Intentional, `tab` component is not fully abstract now (this behavior is needed for `main-tabs`)

Comment (jeluard) [reply src/status_im/components/tabs/views.cljs]: Oops I missed the println ..

**Comment by flexsurfer** [comment]
i thought this is main tabs. ok

File: src/status_im/components/toolbar_new/styles.cljs [hunk 2]
```diff
@@ -90,8 +90,8 @@
    :padding-horizontal 16})
 
 (defstyle item
-  {:ios     {:margin-horizontal 12
-             :margin-vertical   16}
+  {:ios {:margin-horizontal 12
```
Comment (flexsurfer) [inline src/status_im/components/toolbar_new/styles.cljs]: indentation

File: src/status_im/components/toolbar_new/view.cljs [hunk 3]
```diff
@@ -51,8 +51,10 @@
 
 ;; Actions
 
-(defn text-action [handler title]
-  [rn/text {:style tst/toolbar-text-action :on-press handler}
+(defn text-action [{:keys [style handler disabled?]} title]
+  [rn/text {:style    (merge tst/toolbar-text-action style
+                             (if disabled? tst/toolbar-text-action-disabled))
+            :on-press (if-not disabled? handler)}
```
Comment (flexsurfer) [inline src/status_im/components/toolbar_new/view.cljs]: `when-not` ?

File: src/status_im/components/toolbar_new/view.cljs [hunk 4]
```diff
@@ -51,8 +51,10 @@
 
 ;; Actions
 
-(defn text-action [handler title]
-  [rn/text {:style tst/toolbar-text-action :on-press handler}
+(defn text-action [{:keys [style handler disabled?]} title]
+  [rn/text {:style    (merge tst/toolbar-text-action style
+                             (if disabled? tst/toolbar-text-action-disabled))
```
Comment (flexsurfer) [inline src/status_im/components/toolbar_new/view.cljs]: `when` ?

File: src/status_im/ui/screens/wallet/history/views.cljs [hunk 5]
```diff
@@ -167,34 +162,26 @@
...
-                 :tab-list         tabs}]
-     [rn/swiper (merge tst/swiper
-                       {:index (get-tab-index tabs @view-id)
-                        :loop  false})
-                                        ;:ref                    #(reset! swiper %)
-                                        ;:on-momentum-scroll-end (on-scroll-end swiped? scroll-ended @view-id)
+(defn- main-section [view-id tabs ]
+    (let [prev-view-id (r/atom @view-id)]
+    [tabs/swipable-tabs {:style history-styles/main-section
+     ;; TODO also (dispatch [:navigate-to-tab new-view-id ])
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/history/views.cljs]: could you add more info? when should it be uncommented?

Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/history/views.cljs]: and your name

File: src/status_im/ui/screens/wallet/main/views.cljs [hunk 6]
```diff
@@ -38,7 +38,10 @@
   [rn/view {:style wallet-styles/toolbar-buttons-container}
    [vi/touchable-icon :icons/dots-vertical (merge {:color :white} wallet-styles/toolbar-icon) show-not-implemented!]
    [vi/touchable-icon :icons/transaction-history (merge {:color :white :style {:viewBox "-108 65.9 24 24"}} wallet-styles/toolbar-icon)
-    #(rf/dispatch [:navigate-to-modal :wallet-transactions])]])
+    #(rf/dispatch [:navigate-to-modal :wallet-transactions])]
+   (if false
+     ;; TODO placeholder for check on unsigned transactions count
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/main/views.cljs]: could you add your name please

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/components/toolbar_new/view.cljs (hunk 4 above)
Comment (jeluard) [reply src/status_im/components/toolbar_new/view.cljs]: ok

File: src/status_im/components/toolbar_new/view.cljs (hunk 3 above)
Comment (jeluard) [reply src/status_im/components/toolbar_new/view.cljs]: ok

File: src/status_im/ui/screens/wallet/main/views.cljs (hunk 6 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/main/views.cljs]: and this form looks weird, if it always false maybe better to comment it at all?

Comment (jeluard) [reply src/status_im/ui/screens/wallet/main/views.cljs]: That's a placeholder for future features not yet implemented. I will improve the comment.

File: src/status_im/components/toolbar_new/styles.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/components/toolbar_new/styles.cljs]: ok

File: src/status_im/ui/screens/wallet/main/views.cljs (hunk 6 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/main/views.cljs]: thanks

File: src/status_im/ui/screens/wallet/history/views.cljs [hunk 7]
```diff
@@ -167,34 +162,26 @@
...
-    [rn/view {:style history-styles/main-section}
-     [tabs/tabs {:selected-view-id @view-id
-                 :tab-list         tabs}]
-     [rn/swiper (merge tst/swiper
-                       {:index (get-tab-index tabs @view-id)
-                        :loop  false})
-                                        ;:ref                    #(reset! swiper %)
-                                        ;:on-momentum-scroll-end (on-scroll-end swiped? scroll-ended @view-id)
+(defn- main-section [view-id tabs ]
+    (let [prev-view-id (r/atom @view-id)]
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/history/views.cljs]: indentation

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**